app/main.py

```diff
@@ -1,13 +1,25 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return list(
+        map(
+            lambda item: format_single_linter_file(item[0], item[1]),
```

Comment:
You could use unpacking here, e.g. `lambda item: format_single_linter_file(*item)`

Reply:
I did not mean to use `map()` here, but your option is valid. However, your previous code looks more readable which is why prefer the previous form in the future.